oauth2client/contrib/gce.py

```diff
@@ -51,13 +59,17 @@ class AppAssertionCredentials(AssertionCredentials):
     """
 
     @util.positional(2)
-    def __init__(self, scope, **kwargs):
+    def __init__(self, scope='', **kwargs):
         """Constructor for AppAssertionCredentials
 
         Args:
             scope: string or iterable of strings, scope(s) of the credentials
-                   being requested.
+                   being requested. Using this argument will have no effect on
+                   the actual scopes for tokens requested. These scopes are
+                   set at VM instance creation time and won't change.
         """
+        if scope:
+            warnings.warn(_SCOPES_WARNING)
         self.scope = util.scopes_to_string(scope)
```

Comment:
Continuing the "finding ourselves in a hole" discussion from #411: the `scope` attribute now looks silly from a class design point of view in that the class takes it from users, creates a derived product of it that its users could have just as easily created themselves and that may be identical to the original value, and then turns around and sets that derived value on a public instance-scope attribute so that it is available again to the users that supplied it in case they misplaced it. If this were a brand-new class, we'd say to never do that attribute assignment, right? So is that attribute assignment there just to honor existing users who may potentially be out there now making use of the `scope` attribute of their `AppAssertionCredentials` instances? If so, is that worth an implementation comment?

The `scope` instance attribute plays no role in this class' inheritance hierarchy or inherited behavior, right? Right? Pretty please with sugar on top please please please?